tests/topotests/all-protocol-startup/test_all_protocol_startup.py

```diff
@@ -346,6 +347,36 @@ def test_converge_protocols():
     # For debugging after starting FRR/Quagga daemons, uncomment the next line
     ## CLI(net)
 
+def test_nexthop_groups():
+    global fatal_error
+    global net
+
+    # Skip if previous fatal error condition is raised
+    if (fatal_error != ""):
+        pytest.skip(fatal_error)
+
+    print("\n\n** Verifying Nexthop Groups")
+    print("******************************************\n")
+
+    # Create a lib nexthop-group
+    net["r1"].cmd('vtysh -c "c t" -c "nexthop-group red" -c "nexthop 1.1.1.1" -c "nexthop 1.1.1.2"')
+
+    # Create with sharpd using nexthop-group
+    net["r1"].cmd('vtysh -c "sharp install routes 2.2.2.1 nexthop-group red 1"')
+
+    # Verify route and that zebra created NHGs for and they are valid/installed
+    output = net["r1"].cmd('vtysh -c "show ip route 2.2.2.1/32 nexthop-group"')
+    match = re.search(r"Nexthop Group ID: (\d+)", output);
+    assert match is not None, "Nexthop Group ID not found for sharpd route 2.2.2.1/32"
+
+    nhe_id = int(match.group(1))
+
+    output = net["r1"].cmd('vtysh -c "show nexthop-group %d"' % nhe_id)
+    match = re.search(r"Valid", output)
+    assert match is not None, "Nexthop Group ID=%d not marked Valid" % nhe_id
+
+    match = re.search(r"Installed", output)
+    assert match is not None, "Nexthop Group ID=%d not marked Installed" % nhe_id
```

Comment:
so zebra sets an nhg to "installed" and "valid" by default on platforms that don't have real nhg support - is that right? that is, this is correct for both types of target platform?

Reply:
correct